app/db/queries/users.py

```diff
@@ -1,24 +1,82 @@
 from datetime import datetime
+from typing import Optional
 
 from sqlalchemy.orm import Session
 
-from app.models import models, schemas
+from app.models.schemas.users import UserInDB
+from app.models.schemas.memos import MemoCreate
+from app.models import models
 
 
-def create_user(db: Session, user: schemas.UserCreate) -> models.User:
+def create_user(
+    db: Session, username: str, email: str, password: str
+) -> UserInDB:
     now = datetime.now()
-    db_user = models.User(**user.dict(), created_at=now)
+    user = UserInDB(username=username, email=email, created_at=now)
+    user.change_password(password)
+
+    db_user = models.User(**user.dict())
     db.add(db_user)
     db.commit()
     db.refresh(db_user)
-    return db_user
+
+    return user
+
+
+def update_user(
+    db: Session,
+    user: UserInDB,
+    username: Optional[str] = None,
+    email: Optional[str] = None,
+    password: Optional[str] = None,
+) -> UserInDB:
+    org_email = user.email
+    user.username = username or user.username
+    user.email = email or user.email
+    if password:
+        user.change_password(password)
+
+    db.query(models.User).filter(
+        models.User.email == org_email
+    ).update(user.dict())
+    db.commit()
+
+    return user
+
+
+def delete_user_by_email(
+    db: Session,
+    email: str
+) -> bool:
+    row_count = db.query(models.User).filter(
+        models.User.email == email
+    ).delete()
+    if row_count == 1:
+        db.commit()
+        return True
+    else:
+        return False
```

Comment:
This check is no need because email is unique.
```suggestion
    db.commit()
```